contrib/platform/src/com/sun/jna/platform/win32/PdhUtil.java

```diff
@@ -0,0 +1,160 @@
+     *            where the specified performance object or counter is located.
+     *            The computer name can be specified by the DNS name or the IP
+     *            address. If NULL, the function uses the local computer.
+     * @param dwNameIndex
+     *            Index of the performance object or counter.
+     * @return Returns the name of the performance object or counter.
+     */
+    public static String PdhLookupPerfNameByIndex(String szMachineName, int dwNameIndex) {
+        int charToBytes = Boolean.getBoolean("w32.ascii") ? 1 : Native.WCHAR_SIZE;
+
+        // Call once to get required buffer size
+        DWORDByReference pcchNameBufferSize = new DWORDByReference(new DWORD(0));
+        Pdh.INSTANCE.PdhLookupPerfNameByIndex(null, dwNameIndex, null, pcchNameBufferSize);
+
+        // Allocate buffer and call again
+        Memory mem = new Memory(pcchNameBufferSize.getValue().intValue() * charToBytes);
+        Pdh.INSTANCE.PdhLookupPerfNameByIndex(null, dwNameIndex, mem, pcchNameBufferSize);
+
+        // Convert buffer to Java String
+        if (charToBytes == 1) {
+            return mem.getString(0);
+        } else {
+            return mem.getWideString(0);
+        }
+    }
+
+    /**
+     * Utility method to call Pdh's PdhEnumObjectItems that allocates the
+     * required memory for the mszCounterList and mszInstanceList parameters
+     * based on the type mapping used, calls to PdhEnumObjectItems, and returns
+     * the received lists of strings.
+     * 
+     * @param szDataSource
+     *            String that specifies the name of the log file used to
+     *            enumerate the counter and instance names. If NULL, the
+     *            function uses the computer specified in the szMachineName
+     *            parameter to enumerate the names.
+     * @param szMachineName
+     *            String that specifies the name of the computer that contains
+     *            the counter and instance names that you want to enumerate.
+     *            Include the leading slashes in the computer name, for example,
+     *            \\computername. If the szDataSource parameter is NULL, you can
+     *            set szMachineName to NULL to specify the local computer.
+     * @param szObjectName
+     *            String that specifies the name of the object whose counter and
+     *            instance names you want to enumerate.
+     * @param dwDetailLevel
+     *            Detail level of the performance items to return. All items
+     *            that are of the specified detail level or less will be
+     *            returned.
+     * @return Returns a List with two elements. Index 0 contains a List of
+     *         Strings of the counters for the object. Index 1 contains a List
+     *         of Strings of the instances of the object.
+     */
+    public static List<List<String>> PdhEnumObjectItems(String szDataSource, String szMachineName, String szObjectName,
```

Comment:
I would think about splitting this method - is it performance critical? If not, I think two method (one for the conters and one for the instances would be better), the List of List of String feels strange.

Reply:
I had that same thought. In my application I only am using the instance list anyway.   The List of lists is a step up from my original implementation as an array of lists, though, so give me partial credit.